code/ui/components/slider/Slider.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Linq;
+
+using Sandbox;
+using Sandbox.UI;
+using Sandbox.UI.Construct;
+
+namespace TTTReborn.UI
+{
+    /// <summary>
+    /// A horizontal slider. Can be float or whole number.
+    /// </summary>
+    public class Slider : TTTPanel
+    {
+        public Panel Track { get; protected set; }
+        public Panel TrackInner { get; protected set; }
+        public Panel Thumb { get; protected set; }
+
+        /// <summary>
+        /// The right side of the slider
+        /// </summary>
+        public float MaxValue { get; set; } = 100;
+
+        /// <summary>
+        /// The left side of the slider
+        /// </summary>
+        public float MinValue { get; set; } = 0;
+
+        /// <summary>
+        /// If set to 1, value will be rounded to 1's
+        /// If set to 10, value will be rounded to 10's
+        /// If set to 0.1, value will be rounded to 0.1's
+        /// </summary>
+        public float Step { get; set; } = 1.0f;
+
+        public Slider()
```

Comment:
Do we want to make this private/protected? Would we want users to be able to use this constructor?

Reply:
I say leave it.